### DISCUSSION TITLE
RAD - Generate the boilerplate classes from Entity

### DISCUSSION THREAD

**Discussion by fasteddys** [RaythaHQ/raytha#57] [General]
Hello, this is simply awesome, with its simplicity, love the videos as well, I highly encourage others to view them.

Can you help explain, how we can create the various hooks, classes if we have entities, for e.g. EventManagement, with all its supporting needs. Will there be a T4 template or some code gen.
Also whats your road map, and can we use this in a commercial closed source for free

thanks

**Comment by apexdodge** [comment]
Hi @fasteddys thanks for your questions. Glad to hear you like the videos, we have more coming.

> Also whats your road map,

We target two milestones at all times, a minor milestone (currently v0.9.5) and a major milestone (v1.0.0). Minor milestones contain bug fixes and small feature enhancements. Major milestones will be significant enhancements that may require a database migration. Minor releases are released more frequently than major releases.

For v1.0.0 major release: https://github.com/RaythaHQ/raytha/milestone/2 features that are coming include:

- Automatic Headless REST API
- Support for running in a clustered environment (like Azure Kubernetes Service)
- Support for running at a PathBase route (mydomain.com/mysite instead of just mydomain.com)

After that, there's plenty up for discussion as to what the next major release should contain. It will probably be a mix of tackling tech tebt and some of which is listed in Discussions here: https://github.com/RaythaHQ/raytha/discussions. But some low hanging fruit that I am strongly considering in the 3-6 month time frame are:

- Tech debt: https://github.com/RaythaHQ/raytha/issues?q=is%3Aissue+is%3Aopen+label%3A%22tech+debt%22
- Code editor for templates: https://github.com/RaythaHQ/raytha/discussions/17
- New field type for advanced editing: https://github.com/RaythaHQ/raytha/discussions/16
- 301 redirects: https://github.com/RaythaHQ/raytha/discussions/15
- Add text to dashboard: https://github.com/RaythaHQ/raytha/discussions/47

6+ months road map is hard to say, but open to ideas, so feel free to post in the Discussions section. We will likely at some point shift gears to building a "Try it Out" feature on raytha.com to reduce the barrier to playing around with the platform. This will be a precursor to eventually having a Cloud hosted version.

> can we use this in a commercial closed source for free

Yes, Raytha is MIT open source, you can use freely in commercial, closed source products. We offer optional paid support packages, consulting, and hosting services.

> Can you help explain, how we can create the various hooks, classes if we have entities, for e.g. EventManagement, with all its supporting needs. Will there be a T4 template or some code gen.

Can you expand further on this question. Do you mean if you wanted to add more custom database entities (not a Raytha Content Type), what the steps would be?

Thanks

**ps going to transfer this to Discussions board.

**Reply by fasteddys** [reply]
This is great fore thought, let me share my two cents.

- [ ] There's a massive void for developers, an an opportunity area where developers need to produce rapid front-end content from their BusinessObjects. Lot of the other framework come at either/or huge expense and long learning curve. 
- [ ] So auto code genning/generate/scaffolding for the boilerplate, either from an entity or a table first approach, to create the models & controllers and Views would be great. - here is a nice lib... https://github.com/EasyAbp/AbpHelper.GUI
- [ ] Headless is good, but several good CMS have not achieved traction, like Squidex or Orchard then orachrd core headless, with very very long community adoption cycle, and a long learning curve. IMHO, it would really help, if we could create the CRUD views as well. When DNN first came out its adoption was massive because of the ease of creating rich frontend components _QUICKLY_
